crates/store/re_log_types/src/index/absolute_time_range.rs

```diff
@@ -208,6 +213,109 @@ impl AbsoluteTimeRangeF {
+        }
+
+        let min = min.int();
+        let max = max.int();
+
+        if min == max {
+            return self.center(); // We don't care
+        }
+
+        let min_str = min.to_string();
+        let max_str = max.to_string();
+
+        if min_str.len() < max_str.len() {
+            // Different orders of magnitude.
+            // Example: for `61` and `4236`: return 1000
+            return TimeInt::new_temporal(10_i64.pow(max_str.len() as u32 - 1)).into();
+        }
+
+        debug_assert_eq!(min_str.len(), max_str.len());
+
+        // We now have two positive integers of the same length.
+        // We want to find the first non-matching digit,
+        // which we will call the "deciding digit".
+        // Everything before it will be the same,
+        // everything after will be zero,
+        // and the deciding digit itself will be picked as a "smart average"
+        // min:    12345
+        // max:    12780
+        // output: 12500
+
+        let len = min_str.len();
+        for i in 0..len {
+            if min_str.as_bytes()[i] != max_str.as_bytes()[i] {
+                // Found the deciding digit at index `i`
+                let prefix = &min_str[..i];
+                let mut deciding_digit_min = min_str.as_bytes()[i];
+                let deciding_digit_max = max_str.as_bytes()[i];
+
+                debug_assert!(deciding_digit_min < deciding_digit_max);
+
+                let rest_of_min_is_zeroes = min_str.as_bytes()[i + 1..].iter().all(|&c| c == b'0');
+
+                if !rest_of_min_is_zeroes {
+                    // There are more digits coming after `deciding_digit_min`, so we cannot pick it.
+                    // So the true min of what we can pick is one greater:
+                    deciding_digit_min += 1;
+                }
+
+                // if deciding_digit_min == b'1' && rest_of_min_is_zeroes {
+                //     return min; // Special case: the min is just zeroes from here on out. That's perfect!
+                // }
+
+                let deciding_digit = if deciding_digit_min == b'0' {
+                    b'0'
+                } else if deciding_digit_min <= b'5' && b'5' <= deciding_digit_max {
+                    b'5' // 5 is the roundest number in the range
+                } else {
+                    deciding_digit_min.midpoint(deciding_digit_max)
+                };
+
+                let mut result_str = String::with_capacity(len);
+                result_str.push_str(prefix);
+                result_str.push(deciding_digit as char);
+                for _ in i + 1..len {
+                    result_str.push('0');
+                }
+                return TimeInt::new_temporal(
+                    #[expect(clippy::unwrap_used)] // Cannot fail
+                    re_format::parse_i64(&result_str).unwrap(),
+                )
+                .into();
+            }
+        }
+
+        min.into() // All digits are the same
```

Comment:
that should be impossible since we already checked that min and max are different, no?